Client/Frontend/Widgets/AutocompleteTextField.swift

```diff
@@ -136,12 +164,16 @@ class AutocompleteTextField: UITextField, UITextFieldDelegate {
         super.setMarkedText(markedText, selectedRange: selectedRange)
     }
 
-    override func insertText(text: String) {
+    func SELtextDidChange(textField: UITextField) {
         canAutocomplete = true
-        removeCompletion()
-        super.insertText(text)
-        enteredTextLength = count(self.text)
-
+        if completionActive {
+            // As suggestion may not be availaible immediate so we are caching previous
+            // suggestion and if it is valid than we are showing the completion.
```

Comment:
I'd just reduce this to: `// Immediately reuse the previous suggestion if it's still valid.`